From 08f2d46f84e387e6d6de8636b39b103e787d0b57 Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@foss.st.com>
Date: Fri, 3 Sep 2021 13:35:21 +0200
Subject: [PATCH 106/954] feat(st): use stm32_get_otp_value_from_idx() in BL31

The shadowing of the OTP is done in BL2.
As for BL32 on AARCH32 systems, stm32_get_otp_value_from_idx()
should also use bsec_read_otp() in BL31.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 50d48bc656589a6d36aa062a855bf37ef19c169a feat(st): use stm32_get_otp_value_from_idx() in BL31]
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
Change-Id: Ib406cfc154339e6d3cde3c925bc6e9416d77b689
---
 plat/st/common/stm32mp_common.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/plat/st/common/stm32mp_common.c b/plat/st/common/stm32mp_common.c
index 8ebd7ff3e4..72c394de2c 100644
--- a/plat/st/common/stm32mp_common.c
+++ b/plat/st/common/stm32mp_common.c
@@ -192,7 +192,7 @@ int stm32_get_otp_value_from_idx(const uint32_t otp_idx, uint32_t *otp_val)
 
 #if defined(IMAGE_BL2)
 	ret = bsec_shadow_read_otp(otp_val, otp_idx);
-#elif defined(IMAGE_BL32)
+#elif defined(IMAGE_BL31) || defined(IMAGE_BL32)
 	ret = bsec_read_otp(otp_val, otp_idx);
 #else
 #error "Not supported"
-- 
2.53.0

